ydb/src/grpc_wrapper/raw_topic_service/common/codecs.rs

```diff
@@ -1,8 +1,8 @@
 use itertools::Itertools;
 use ydb_grpc::ydb_proto::topic::{Codec, SupportedCodecs};
 
-#[derive(serde::Serialize)]
-pub(crate) enum RawCodec {
+#[derive(serde::Serialize, Clone)]
+pub enum RawCodec {
```

Comment:
since I am using this struct in a CreateTopicOptions, pub(create) mod had to be changed. I am not quite sure if this a good thing that Raw struct becomes pub. I just didn't want to copy-paste another struct just to put in user-space level i guess...

Reply:
I think in the case copy-paste is good:

1. When we strong separate public interface from internals - it will easy to make change internals in the future. 
2. It prevent public some private API (for example internal errors from methods, From trait implementations, may be some methods).


And for public interface will be better have no strung difference between well-known and custom codecs: for better add codecs in the future.
And in code work with custom and well-known code will same (may be optional support default-implementation for well-known codec, but may be not for all and customer can re-define it).

```rust
struct Codec {
    pub code: i32
}

impl Codec {
    pub const RAW: Codec = Codec{code:1};
    pub const GZIP: Codec = Codec{code:2};

    pub fn is_custom(Codec: Codec)->bool{
        Codec.code >= 10000 && Codec.code < 20000
    }
}
```